client/src/components/PlayerPage/index.tsx

```diff
@@ -1,18 +1,83 @@
-import { useEffect } from "react";
+import { useEffect, useState } from "react";
+import Button from '@mui/material/Button';
+import ButtonGroup from '@mui/material/ButtonGroup';
+import Grid from '@mui/material/Grid';
+import CircularProgress from '@mui/material/CircularProgress';
 
-interface IPlayerPageProps { }
+import gameService from "../../services/gameService";
+import socketService from "../../services/socketService";
 
-export function PlayerPage(props: IPlayerPageProps) {
+export function PlayerPage() {
+
+    const [isLoading, setIsLoading] = useState(false);
+    const [question, setQuestion] = useState(['None']);
+    const [hasAnswered, setHasAnswered] = useState(false);
+    const [waitingText, setWaitingText] = useState("Waiting for game to start...");
+
+    // Listen for the player join event from roomService and update our state if one joins
+    const handleNewQuestion = () => {
+        if (socketService.socket)
+            gameService.onSendQuestion(socketService.socket, (question) => {
+                console.log(question);
+                setQuestion(question);
+            });
+    };
+
+    // This function is called when the user presses the "host" button
+    const sendAnswer = async (e: React.FormEvent) => {
+
+        // Prevent the page from refreshing
+        e.preventDefault();
+
+        setIsLoading(true);
+
+        const answerIndex = e.currentTarget.id;
+        console.log(answerIndex);
+
+        // Send the answer to the server
+        const socket: any = socketService.socket;
+        const response = await gameService.sendAnswer(socket, answerIndex).catch((err) => {
+            alert(err);
+        });
+
+        console.log(response);
+
+        setQuestion(['None']);
+        setWaitingText('Waiting for others to answer...');
```

Comment:
Does waiting text ever change from 'Waiting for others to answer...'? If not, can't we just make it a normal const variable?

Reply:
It could change to waiting for winners decision or waiting for round to start?